File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.java
import java.util.*;

class Solution {
    
    public int solution(int[] elements) {
        int answer = 0;
        Set<Integer> s = new HashSet<>();
        
        int len = elements.length;
        
        for(int i=1; i<len+1; i++){ // i = 자를 길이
            int idx = 0;
            // 1번째 원소부터 i번째 원소까지 자르기
            int[] nums = Arrays.copyOfRange(elements, idx, i);
            // 자른 배열의 총합
            int sum = Arrays.stream(nums).sum();
            s.add(sum);
            
            idx += 1;
            while (idx != 0){
                sum -= elements[idx-1];
                sum += elements[(idx+i-1)%len];
                s.add(sum);
                
                
                idx  = (idx+1)%len;
            }
        }
        
        return s.size();
    }
}

File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.py
def solution(elements):
    answer = 0
    arr = []
    n = len(elements)
    memory = []
    
    for i in range(n):
        tmp = sum(elements[:i+1])
        memory.append(tmp)
        
        for j in range(n):
            tmp -= elements[j] 
            tmp += elements[(j+i+1)%n]
            
            memory.append(tmp)
        
        
    answer = len(set(memory))
    return answer